Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -218,79 +149,49 @@ public async Task PreProc_EndToEnd_SuccessCase_With_Organization()
         public async Task PreProc_EndToEnd_ShouldWarnOrNot_SonarQubeDeprecatedVersion(string sqVersion, bool shouldWarn)
         {
             // Arrange
-            var workingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
-            var logger = new TestLogger();
-
-            var mockServer = MockSonarQubeServer();
-            mockServer.Data.SonarQubeVersion = new Version(sqVersion);
-            var mockAnalyzerProvider = MockAnalyzerProvider();
-            var mockTargetsInstaller = new Mock<ITargetsInstaller>();
-            var mockFactory = new MockObjectFactory(mockServer, mockTargetsInstaller.Object, mockAnalyzerProvider);
-
-            using (PreprocessTestUtils.CreateValidNonTeamBuildScope())
-            using (new WorkingDirectoryScope(workingDir))
-            {
-                var settings = TeamBuildSettings.GetSettingsFromEnvironment(new TestLogger());
-                settings.Should().NotBeNull("Test setup error: TFS environment variables have not been set correctly");
-                settings.BuildEnvironment.Should().Be(BuildEnvironment.NotTeamBuild, "Test setup error: build environment was not set correctly");
-
-                var preProcessor = new PreProcessor(mockFactory, logger);
+            using var scope = new TestScope(TestContext);
+            var factory = new MockObjectFactory();
+            factory.Server.Data.SonarQubeVersion = new Version(sqVersion);
+            var preProcessor = new PreProcessor(factory, factory.Logger);
 
-                // Act
-                var success = await preProcessor.Execute(CreateArgs());
-                success.Should().BeTrue("Expecting the pre-processing to complete successfully");
-            }
+            var success = await preProcessor.Execute(CreateArgs());
+            success.Should().BeTrue("Expecting the pre-processing to complete successfully");
 
-            mockTargetsInstaller.Verify(x => x.InstallLoaderTargets(workingDir), Times.Once());
+            factory.TargetsInstaller.Verify(x => x.InstallLoaderTargets(scope.WorkingDir), Times.Once());
 
             if (shouldWarn)
             {
-                mockServer.AssertWarningWritten("version is below supported");
+                factory.Server.AssertWarningWritten("version is below supported");
             }
             else
             {
-                mockServer.AssertNoWarningWritten();
+                factory.Server.AssertNoWarningWritten();
             }
         }
 
         [TestMethod]
         public async Task PreProc_NoPlugin()
         {
             // Arrange
```

Comment:
I think we should either delete all of them or use comments for // Act and // Assert too.